### PR TITLE
Convert ShadedJar into artifact transform

### PR DESCRIPTION
**Pull request by wolfs**

### CODE REVIEW

File: buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTestingPlugin.kt [hunk 1]
```diff
@@ -67,22 +69,24 @@ class DistributionTestingPlugin : Plugin<Project> {
...
+    fun Project.configureGradleTestEnvironment(distributionTest: DistributionTest): Unit = distributionTest.run {
         gradleInstallationForTest.run {
             val intTestImage: Sync by tasks
-            val toolingApiShadedJar: Zip by rootProject.project(":toolingApi").tasks
+            val toolingApiShadedJar: ShadedJar by rootProject.project(":toolingApi").tasks
             gradleHomeDir.set(dir { intTestImage.destinationDir })
             gradleUserHomeDir.set(rootProject.layout.projectDirectory.dir("intTestHomeDir"))
             daemonRegistry.set(rootProject.layout.buildDirectory.dir("daemon"))
-            toolingApiShadedJarDir.set(dir { toolingApiShadedJar.destinationDir })
+            toolingApiShadedJarDir.set(dir { toolingApiShadedJar.jarFile.get().asFile })
```
Comment (lptr) [inline buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTestingPlugin.kt]: This doesn't make sense. a) we are connecting two providers, but we are doing it via a closure, b) we are setting a directory property to a directory that points to a file?!

Comment (lptr) [reply buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTestingPlugin.kt]: How is b) not result in an error somewhere?

Comment (wolfs) [reply buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTestingPlugin.kt]: I actually want to set this to the `parentFile` of the `jarFile`. I don't know how the tests passed with this code...

Comment (wolfs) [reply buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTestingPlugin.kt]: Ahh, I set it in the wrong place - `DistributionTest`...

Comment (wolfs) [reply buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTestingPlugin.kt]: But you are right, validation should catch this!

Comment (wolfs) [reply buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTestingPlugin.kt]: Fixed.

**Comment by wolfs** [comment]
Replaced by https://github.com/gradle/gradle/pull/4820.

**Pull request closed without merging**